.github/workflows/release.yml

```diff
@@ -0,0 +1,72 @@
+name: Build and publish release
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
I'd have to brush up on gh actions, but this looks like it will execute on push to `main` as well as the tag definition you supplied. Seems like you may not want the `main` branch referenced here